src/main/java/duke/command/UpdateTaskCommand.java

```diff
@@ -37,7 +40,17 @@ public UpdateTaskCommand(String command) {
     @Override
     public void execute(PatientTaskList patientTask, TaskManager taskManager, PatientManager patientManager,
                         Ui ui, PatientTaskStorage patientTaskStorage,
-                        TaskStorage taskStorage, PatientStorage patientStorage) throws DukeException {
+                        TaskStorage taskStorage, PatientStorage patientStorage, CmdFreqStorage cmdFreqStorage,
+                        CommandManager commandManager) throws DukeException {
+        this.hasBeenAddedBefore = true;
+        String commandName = this.getClass().getSimpleName();
+        if (!hasBeenAddedBefore) {
+            commandManager.getCmdFreqTable().put(commandName, 1);
+        }
+        int count = commandManager.getCmdFreqTable().containsKey(commandName)
+                    ? commandManager.getCmdFreqTable().get(commandName) : 0;
+        commandManager.getCmdFreqTable().put(commandName, count + 1);
+
```

Comment:
All these are not belong to command.execute. 
1.  The condition check for <FreqCommandHandler.hasBeenAddedBefore(getClass().getName())>
2.  The increment of count

There should be a specific class to handle. Do an abstraction.

Reply:
I have shifted all of these to a function in our abstract command class.